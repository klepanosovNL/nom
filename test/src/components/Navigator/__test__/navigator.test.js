import React from 'react';
import renderer from 'react-test-renderer';
import { Navigator } from '../Navigator';
import { mockStore, wrapProvider } from '../../../utils/mockedStore';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';

jest.mock('../../../store/actionCreators', () => ({
	setCurrentPreset: (...args) => ({ type: 'TEST_TYPE', args }),
	toggleDisabledItems: (...args) => ({ type: 'TEST_TYPE_TOGGLE', args }),
	toggleDisabledItemsByName: (args) => ({
		type: 'TEST_TYPE_TOGGLE_DISABLED',
		args,
	}),
}));

describe('Navigator"s tests', () => {
	const initialState = {
		list: [],
		currentPreset: 'none',
		filterByCategories: '',
		filterByNameInput: '',
		blockedButton: '',
		customs: [],
	};

	const store = mockStore(initialState);

	it('Check event: clickHandler(name)', async () => {
		render(wrapProvider(<Navigator></Navigator>, store));

		const label = screen.getByText(/low/i);
		fireEvent.click(label);

		await waitFor(() => {
			expect(store.getActions()).toEqual([
				{
					type: 'TEST_TYPE',
					args: ['low'],
				},
				{
					type: 'TEST_TYPE_TOGGLE',
					args: [false],
				},
				{
					type: 'TEST_TYPE_TOGGLE_DISABLED',
					args: ['Literature', 'Software', 'Dogs', 'Cartoons'],
				},
			]);
		});
	});

	it('Navigator"s snapshot', () => {
		const tree = renderer
			.create(wrapProvider(<Navigator></Navigator>, store))
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
