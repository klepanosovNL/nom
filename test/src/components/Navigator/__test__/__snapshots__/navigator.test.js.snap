// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator"s tests Navigator"s snapshot 1`] = `
<div
  className="presets"
>
  <button
    className="btn_default presets__item presets__item_selected"
    onClick={[Function]}
  >
    none
  </button>
  <button
    className="btn_default presets__item "
    onClick={[Function]}
  >
    low
  </button>
  <button
    className="btn_default presets__item "
    onClick={[Function]}
  >
    strong
  </button>
  <button
    className="btn_default presets__item "
    onClick={[Function]}
  >
    custom
  </button>
</div>
`;
